jupyterhub/files/hub/jupyterhub_config.py

```diff
@@ -60,10 +60,15 @@ def camelCaseify(s):
 
 # configure the hub db connection
 db_type = get_config("hub.db.type")
+db_password = get_secret_value("hub.db.password", None)
 if db_type == "sqlite-pvc":
     c.JupyterHub.db_url = "sqlite:///jupyterhub.sqlite"
 elif db_type == "sqlite-memory":
     c.JupyterHub.db_url = "sqlite://"
+elif db_type == "mysql" and db_password != None:
+    os.environ["MYSQL_PWD"] = db_password
+elif db_type == "postgres" and db_password != None:
+    os.environ["PGPASSWORD"] = db_password
```

Comment:
I don't think these want to be in the same if/elif branches for db_url. This code as it is would mean `hub.db.url` is not set if db_type is mysql or postgres with a password, which I think isn't correct (generally db_url *must* be set in these cases). A separate if/elif block for setting the password env should be right.

Reply:
Yikes yes very true, happy about your sharp eyes catching this!

Reply:
Resolved by 5f8fac2